examples/text-generation/utils.py

```diff
@@ -429,8 +429,13 @@ def setup_distributed_model(args, model_dtype, model_kwargs, logger):
 
     logger.info("DeepSpeed is enabled.")
     deepspeed.init_distributed(dist_backend="hccl")
-    config = AutoConfig.from_pretrained(args.model_name_or_path, torch_dtype=model_dtype, **model_kwargs)
```

Comment:
We cannot remove `torch_dtype=model_dtype` here and then use `config.torch_dtype` afterwards. Because we won't be able to specify anymore if we want to run the model in fp32 or in bf16.
Why do you need to do that?

Reply:
It seems for distributed model, we only support deepspeed, and whenever deepspeed is on, it's aways bf16 or fp8.  I wonder why we have a limitation like that. 

@jaygala223  Any specific reason why you have to change to config.torch_dtype?   What issue were you seeing with fp8?  Also, all other code setup_model, setup_model_distributed_ep/tp are still using model_dtype by the way.